activejob/lib/active_job/queue_adapters.rb

```diff
@@ -48,5 +48,32 @@ module QueueAdapters
     autoload :SneakersAdapter
     autoload :SuckerPunchAdapter
     autoload :TestAdapter
+
+    ADAPTER = 'Adapter'.freeze
+
+    class << self
+      def instance(name, *adapter_args, &block)
+        hash = ([name, block] << adapter_args).hash
```

Comment:
Oh, I skipped that part. Agree with @sgrif though - I think this argument is absurd.